File: deleteUsers/handler.py
import boto3, json, os

client = boto3.resource("dynamodb")

IS_OFFLINE = os.getenv("IS_OFFLINE", False)
if IS_OFFLINE:
    boto3.Session(
        aws_access_key_id=os.getenv("AWS_ACCESS_KEY_ID", "ACCESS_KEY"),
        aws_secret_access_key=os.getenv("AWS_SECRET_ACCESS_KEY", "SECRET_KEY"),
    )
    client = boto3.resource("dynamodb", endpoint_url="http://localhost:8000")

table = client.Table("usersTable")


def deleteUsers(event, context):
    user_id = event["pathParameters"]["id"]
    result = table.delete_item(Key={"pk": user_id})

    body = json.dumps({"message": f"User {user_id} deleted"})

    response = {
        "statusCode": result["ResponseMetadata"]["HTTPStatusCode"],
        "body": body,
    }

    return response

File: updateUsers/handler.js
const aws = require("aws-sdk")

let dynamoDBClientParams = {}

if (process.env.IS_OFFLINE) {
    dynamoDBClientParams =  {
        region: 'localhost',
        endpoint: 'http://localhost:8000',
        accessKeyId: process.env.AWS_ACCESS_KEY_ID,
        secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY
    }
}

const dynamodb = new aws.DynamoDB.DocumentClient(dynamoDBClientParams)

const updateUsers = async (event, context) => {

    let userId = event.pathParameters.id

    const body = JSON.parse(event.body)

    var params = {
        TableName: 'usersTable',
        Key: { pk: userId },
        UpdateExpression: 'set #age = :age, #address = :address',
        ExpressionAttributeNames: { '#age' : 'age', '#address' : 'address'},
        ExpressionAttributeValues:
            { ':age' : body.age, ':address' : body.address},
        ReturnValues: 'ALL_NEW'
    };

    return dynamodb.update(params).promise().then(res => {
        console.log(res)
        return {
            "statusCode": 200,
            "body": JSON.stringify({ 'user': res.Attributes })
        }
    })
}

module.exports = {
    updateUsers
}